File: languages/FSML/Java/org/softlang/fluent/Sample.java
// BEGIN ...
package org.softlang.fluent;

import static org.softlang.fluent.FsmlImpl.fsm;

public class Sample {
	
	public static final
// END ...
		Fsml sample =
			fsm()
				.state("locked")
					.transition("ticket", "collect", "unlocked")
					.transition("pass", "alarm", "exception")
				.state("unlocked")
					.transition("ticket", "eject", "unlocked")
					.transition("pass", null, "locked")
				.state("exception")
					.transition("ticket", "eject", "exception")
					.transition("pass", null, "exception")
					.transition("mute", null, "exception")
					.transition("release", null, "locked");
// BEGIN ...
}
// END ...

File: languages/FSML/Haskell/Language/FSML/StringTemplate/Sample.hs
-- BEGIN ...
module Language.FSML.StringTemplate.Sample where
import Language.FSML.StringTemplate.Syntax
-- END ...
sampleFsm :: Fsm
sampleFsm =
 Fsm [
  State {
   initial=True,
   stateid="locked",
   transitions=[
    Transition { event="ticket", action=Just "collect", target=Just "unlocked" },
    Transition { event="pass", action=Just "alarm", target=Just "exception" }
   ]
  },
  State {
   initial=False,
   stateid="unlocked",
   transitions=[
    Transition { event="ticket", action=Just "eject", target=Nothing },
    Transition { event="pass", action=Nothing, target=Just "locked" }
   ]
  },
  State {
   initial=False,
   stateid="exception",
   transitions=[
    Transition { event="ticket", action=Just "eject", target=Nothing },
    Transition { event="pass", action=Nothing, target=Nothing },
    Transition { event="mute", action=Nothing, target=Nothing },
    Transition { event="release", action=Nothing, target=Just "locked" }
   ]
  }
 ]
